datacapture/src/main/java/com/google/android/fhir/datacapture/views/QuestionnaireItemDatePickerViewHolderFactory.kt

```
@@ -50,7 +50,13 @@ object QuestionnaireItemDatePickerViewHolderFactory : QuestionnaireItemViewHolde
 
           // TODO: find a more robust way to do this as it is not guaranteed that the activity is
           // an AppCompatActivity.
-          val context = (itemView.context as ContextThemeWrapper).baseContext as AppCompatActivity
+          // The application is wrapped in a ContextThemeWrapper in QuestionnaireFragment and again
+          // in TextInputEditText during layout inflation. As a result, it is necessary to
+          // access the base context twice to retrieve the application object from the view's
+          // context.
+          val context =
```

There's no guarantee that the base context is an `AppCompatActivity` or even an `Activity`. Also, you might want to take a look at this [function](https://cs.android.com/androidx/platform/frameworks/support/+/androidx-master-dev:appcompat/appcompat/src/main/java/androidx/appcompat/app/AppCompatDelegateImpl.java;l=1615).

---

Not much tbh, I'd probably throw an exception.